packages/react-flicking/src/react-flicking/Flicking.tsx

```diff
@@ -0,0 +1,191 @@
+import NativeFlicking, { FlickingOptions, FlickingPanel, FlickingStatus, Plugin, FlickingEvent, NeedPanelEvent } from "@egjs/flicking";
+import * as React from "react";
+import { findDOMNode } from "react-dom";
+import ChildrenDiffer from "@egjs/react-children-differ";
+import { CloneComponent } from "./Clone";
+import { DEFAULT_OPTIONS, FLICKING_PROPS } from "./consts";
+import { FlickingProps, FlickingType } from "./types";
+import ListDiffer from "@egjs/list-differ";
+import { ChildrenDiffResult } from "@egjs/children-differ";
+
+export default class Flicking extends React.Component<Partial<FlickingProps & FlickingOptions>> implements FlickingType<Flicking> {
+  public static defaultProps: FlickingProps = FLICKING_PROPS;
+  // Flicking
+  public options: FlickingOptions = {
+    ...DEFAULT_OPTIONS,
+    autoResize: true,
+  };
+  public state: {
+    cloneCount: number,
+  } = {
+      cloneCount: 0,
+    };
+  // Flicking
+  private flicking?: NativeFlicking | null;
+  // differ
+  private differ?: ChildrenDiffer;
+  private pluginsDiffer: ListDiffer<Plugin> = new ListDiffer<Plugin>();
+  // life cycle
+  public render() {
+    const props = this.props;
+    // tslint:disable-next-line:naming-convention
+    const Tag = props.tag as any;
+    const options = this.options;
+    const attributes: { [key: string]: any } = {};
+
+    for (const name in props) {
+      if (name in DEFAULT_OPTIONS) {
+        options[name] = props[name];
+      } else if (!(name in FLICKING_PROPS)) {
+        attributes[name] = props[name];
+      }
+    }
+    return (
+      <Tag {...attributes}>
+        <div className="eg-flick-viewport">
+          <div className="eg-flick-camera">
+            <ChildrenDiffer ref={e => { this.differ = e as ChildrenDiffer; }}>
+              {this.renderPanels()}
+            </ChildrenDiffer>
+          </div>
+        </div>
+      </Tag>
+    );
+  }
+  public componentDidUpdate() {
+    const result = this.differ!.update();
+
+    this.flicking!.sync(result as ChildrenDiffResult<HTMLElement>);
+    this.checkPlugins();
+    this.checkCloneCount();
+    if (this.props.lastIndex! >= -1) {
```

Comment:
I hope this to be `typeof this.props.lastIndex! === "number"`.
As restricting lastIndex value to -1 is not a part of how NativeFlicking works.